lib/saml_idp.rb

```diff
@@ -9,17 +9,26 @@ module SamlIdp
   require 'saml_idp/metadata_builder'
   require 'saml_idp/version'
   require 'saml_idp/engine' if defined?(::Rails) && Rails::VERSION::MAJOR > 2
+  require 'request_store'
 
   def self.config
-    @config ||= SamlIdp::Configurator.new
+    if RequestStore.store[:request_config]
+      RequestStore.store[:request_config]
+    else
+      raise "Config must be set before processing the request"
+    end
   end
 
   def self.configure
     yield config
   end
 
   def self.metadata
-    @metadata ||= MetadataBuilder.new(config)
+    MetadataBuilder.new(config)
```

Comment:
Not sure how performant `MetadataBuilder` is but if it's heavy we can use a `RequestStore` variable and make it a request-level singleton.

Reply:
Just took a look at MetadataBuilder seems like the constructor is just setting an instance variable so we're good.

Reply:
Yeah I thought about that, but I wanted to make minimal changes and I thought that metadata is something that is not requested very frequently. Usually on requested during setup.

Do you think I should change to request store though?

Like this?
```ruby
  def self.metadata
    RequestStore.store[:metadata] ||= MetadataBuilder.new(config)
  end
```

Reply:
No, let's keep it as it is. We should monitor the performance and if it was problematic we can update it.